ws/src/main/scala/de/thm/ii/fbs/services/conferences/BBBService.scala

```diff
@@ -25,25 +27,25 @@ class BBBService(templateBuilder: RestTemplateBuilder,
                  @Value("${services.bbb.service-url}") private val apiUrl: String,
                  @Value("${services.bbb.shared-secret}") private val secret: String)
   extends ConferenceService {
-  /**
-    * The name of the conference service
-    */
-  val name = "bigbluebutton"
-
   /**
     * Creates a new Conference using BBB
-    * @param cid the id for the new conference
+    * @param courseId the id for the new conference
     * @return the newly created conference
     */
-  override def createConference(cid: String): Conference = {
+  override def createConference(courseId: String): Conference = {
+    val cid = UUID.randomUUID().toString
     val participantPassword = UUID.randomUUID().toString
     val modPassword = UUID.randomUUID().toString
+
+    // actual registering of conference against BBB api
     this.registerBBBConference(cid, cid, participantPassword, modPassword)
     new Conference {
       override val id: String = cid
-      override val serviceName: String = BBBService.this.name
-      private val meetingPassword = participantPassword
-      private val moderatorPassword = modPassword
+      override val serviceName: String = BBBService.name
+      override val visibility: String = "false"
```

Comment:
This must be a var because `visibility` in Conference is a var.

Reply:
Why is visibility a String?